main_test.go

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package section is responsible for splitting a program into its different
+// sections without attempting to parse the contents.s
+package main
+
+import (
+	"bytes"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/pkg/diff"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Arguments struct {
+	args []string
+	fset *token.FileSet
+}
+
+var command = Arguments{
+	args: []string{"-l", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"},
+	fset: token.NewFileSet(),
+}
```

Comment:
I'm not sure why we need this.
The token.FileSet should not be shared between tests.
command.args is used 4 times. We can just hard code the values we care about as needed.

Plus this is the global scope—shared with every other .go file in this directory.
"Arguments" and "command" are very general terms that might have other meanings in other .go files.
If we really needed globals here, we would want to name them something a lot more specific given how they're used.